specs/Popup-AdditionalLayoutProperties-Spec.md

```diff
@@ -0,0 +1,218 @@
+the popup.  This will allow elements such as `CommandBarFlyout` to be able to rely on system XAML for the placement of their
+child popups in a way that will take monitor or app bounds into account without needing to do that accounting manually.
+
+## Visual Examples
+
+**Opening down**
+
+When CommandBarFlyout has enough space below its primary commands, we want it to open down.
+
+Secondary commands flyout closed:
+
+![Shows the CommandBarFlyout not yet open with sufficient monitor space below it](images/CommandBarFlyout-SufficientSpace.png)
+
+Secondary commands flyout open:
+
+![Shows the CommandBarFlyout opened down](images/CommandBarFlyout-SufficientSpace-Open.png)
+
+**Opening up**
+
+When CommandBarFlyout does *not* have enough space below its primary commands, we want it to be able to open up.
+
+Secondary commands flyout closed:
+
+![Shows the CommandBarFlyout not yet open with insufficient monitor space below it](images/CommandBarFlyout-InsufficientSpace.png)
+
+Secondary commands flyout open:
+
+![Shows the CommandBarFlyout opened up](images/CommandBarFlyout-InsufficientSpace-Open.png)
+
+# API Pages
+
+## Popup class
+
+```csharp
+class Popup
+{
+    // Existing APIs
+    // ...
+
+    // New APIs
+    UIElement PlacementTarget { get; set; }
+    PopupPlacementMode DesiredPlacement { get; set; }
+    PopupPlacementMode ActualPlacement { get; }
+    
+    public event System.EventHandler<object> PlacementChanged;
+}
+```
+
+The example below shows how the `Placement` APIs are used by the 
+[CommandBarFlyout](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.CommandBarFlyout)
+to position its
+[CommandBarFlyoutCommandBar](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.CommandBarFlyoutCommandBar)'s
+secondary commands Popup, and how to respond to the event raised when XAML places the Popup.
+
+```xml
+<!-- Part of the CommandBarFlyoutCommandBar's default template -->
+<Popup
+    x:Name="OverflowPopup"
+    PlacementTarget="{Binding ElementName=PrimaryItemsRoot}"
+    DesiredPlacement="Bottom">
+</Popup>
+```
+
+```csharp
+void OnApplyTemplate()
+{
+    m_overflowPopup = GetTemplateChild("OverflowPopup");
+    m_overflowPopup.PlacementChanged += OnOverflowPopupPlacementChanged;
+}
+
+void OnOverflowPopupPlacementChanged(object sender, object args)
+{
+    UpdateVisualState(useTransitions: false);
+}
+
+void UpdateVisualState(bool useTransitions)
+{
+    if (m_overflowPopup.ActualPlacement == PopupPlacementMode.Top)
+    {
+        VisualStateManager.GoToState(this, "ExpandedUp", useTransitions);
+    }
+    else
+    {
+        VisualStateManager.GoToState(this, "ExpandedDown", useTransitions);
+    }
+}
+```
+
+## Popup.PlacementTarget property
+
+Use this property to describe which element the `Popup` should be positioned relative to.
+Defaults to `null`.
+
+If this is `null`, then `DesiredPlacement` is ignored, `ActualPlacement` is always `None`, and
```

Comment:
`None` => `Auto`? (Here and elsewhere)

Reply:
I prefer None to Auto as well.  @MikeHillberg are you OK with None, or do you still want Auto?

Reply:
I didn't want `None`, because placement of None sounds like it won't open.

I also like `Auto` because that's what [FlyoutPlacementMode](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutPlacementMode) uses. (It doesn't have the complication of an Actual property though.)

I think returning Auto from the Actual is less confusing than None. Auto doesn't tell you anything, None seems to. Auto tells you that the system decided where to put it, it's just not telling you what it decided.

Reply:
OK, I'm cool with `Auto` if you feel strongly about it.